lib/kubeclient.rb

```diff
@@ -589,6 +598,16 @@ def api
                                          security_context_constraints]
     }.freeze
 
+    def build_http_conflict_reason(e)
```

Comment:
It's unfortunate that we're parsing the JSON response twice (once in `build_http_error_message` and once here during a 409) but I noticed `build_http_error_message` was a public method so I wanted to avoid making a breaking change. I don't have much context on why it's public, but perhaps in the future it should also be `private`?

Reply:
My stance for quite a while has been that if it's not documented in the README, it's internal and we're free to break it, even if it's not `private` on the ruby level.
So feel free to change this, and/or make it `private`.

Reply:
Speaking of which, please document `ResourceAlreadyExistsError` in README.  And if you don't mind, `ResourceNotFoundError` which I see we forgot.

Double parsing is suboptimal but not deal-breaker, I'm OK with merging as-is if you prefer.  Otherwise LGTM :100:, thanks for this PR!

Reply:
> Speaking of which, please document `ResourceAlreadyExistsError` in README. And if you don't mind, `ResourceNotFoundError` which I see we forgot.

Ohhh yeah, good call!

Do you have a preference on where to put these?

Perhaps a section `### Errors` within `## Usage`? Perhaps after `### Timeouts`?